src/Microsoft.DotNet.DeadRegionAnalysis.Tests/PreprocessorExpressionEvaluatorTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace Microsoft.DotNet.DeadRegionAnalysis.Tests
+{
+    public class PreprocessorExpressionEvaluatorTests
+    {
+        [Fact]
+        public void TrueLiteral()
+        {
+            Verify("true", Tristate.True);
+        }
+
+        [Fact]
+        public void FalseLiteral()
+        {
+            Verify("false", Tristate.False);
+        }
+
+        [Fact]
+        public void LogicalAnd()
+        {
+            Verify("true && true", Tristate.True);
+            Verify("true && false", Tristate.False);
+            Verify("true && varying", Tristate.Varying);
+            Verify("false && false", Tristate.False);
+            Verify("false && varying", Tristate.False);
+        }
+
+        [Fact]
+        public void LogicalOr()
+        {
+            Verify("true || true", Tristate.True);
+            Verify("true || false", Tristate.True);
+            Verify("true || varying", Tristate.True);
+            Verify("false || false", Tristate.False);
+            Verify("false || varying", Tristate.Varying);
+        }
+
+        [Fact]
+        public void LogicalNot()
+        {
+            Verify("!true", Tristate.False);
+            Verify("!false", Tristate.True);
+            Verify("!varying", Tristate.Varying);
+        }
+
+        [Fact]
+        public void ParentheicalExpressions()
+        {
+            Verify("!(true && (false || true))", Tristate.False);
+            Verify("(!((true)))", Tristate.False);
+            Verify("(!(!true))", Tristate.True);
+        }
+
+        [Fact]
+        public void InvalidExpressions()
+        {
+            VerifyInvalid("false + true");
+            VerifyInvalid("&varying");
+            VerifyInvalid("1");
+        }
+
+        [Fact]
+        public void OverrideLiteralExpressions()
+        {
+            var evaluator = new PreprocessorExpressionEvaluator(
+                new Dictionary<string, Tristate>()
+                {
+                    { "true", Tristate.False },
+                    { "false", Tristate.True }
+                },
+                Tristate.Varying);
+
+            Assert.Equal(Tristate.False, EvaluateExpression("true", evaluator));
+            Assert.Equal(Tristate.True, EvaluateExpression("false", evaluator));
+        }
+
+        private static readonly CSharpParseOptions s_expressionParseOptions = new CSharpParseOptions(
+            documentationMode: DocumentationMode.None,
+            kind: SourceCodeKind.Interactive);
+
+        private static readonly PreprocessorExpressionEvaluator s_evaluator = new PreprocessorExpressionEvaluator(
```

Comment:
Why static here?  This is a mutable type from what I can tell.  I would make it instance state.  Making it static means we have to proactively disable xUnit features like running in parallel.